crates/sync/src/spsc.rs

```
@@ -0,0 +1,161 @@
+        let index =
+            self.buf.head.load(Ordering::Relaxed) % self.buf.capacity;
+        unsafe {
+            // Elide mutablity
+            let mut b = self.buf.buffer;
+            b.as_mut()[index] = item;
+        }
+        self.buf.head.fetch_add(1, Ordering::Relaxed);
+        Some(())
+    }
+
+    pub unsafe fn inner(&self) -> &SpscRingBuffer<T> {
```

Clippy (1.96.0)

[warning] unsafe function's docs are missing a `# Safety` section

(warning)